countItems/counter.js

```diff
@@ -0,0 +1,122 @@
+const async = require('async');
+const werelogs = require('werelogs');
+const { BucketInfo } = require('arsenal').models;
+const { reshapeExceptionError } = require('arsenal').errorUtils;
+
+const loggerConfig = {
+    level: 'debug',
+    dump: 'error',
+};
+
+werelogs.configure(loggerConfig);
+const log = new werelogs.Logger('S3Utils::CountItems::Worker');
+
+let mongoClient = require('./mongoClient')();
+
+function clientSetup(callback) {
+    if (mongoClient.client && mongoClient.client.isConnected()) {
+        log.debug('mongoclient is connected...skipping setup');
+        return callback();
+    }
+    return mongoClient.setup(callback);
+}
+
+function countItems(bucketInfoObj, callback) {
+    const bucketInfo = BucketInfo.fromObj(bucketInfoObj);
+    const bucketName = bucketInfo.getName();
+    log.debug(`${process.pid} handling ${bucketName}`);
+    async.waterfall([
+        next => mongoClient._getIsTransient(bucketInfo, log, next),
+        (isTransient, next) => mongoClient.getObjectMDStats(
+            bucketName, bucketInfo, isTransient, log, next),
+    ], callback);
+}
+
+function clientTeardown(callback) {
+    if (mongoClient.client && mongoClient.client.isConnected()) {
+        log.debug('mongoclient is connected...closing client');
+        return mongoClient.close(callback);
+    }
+    return callback();
+}
+
+let ranOnce = false;
+function handleTestMessages(data) {
```

Comment:
@alexanderchan-scality I realize this is still WIP but better catch it early: please avoid test-related code with "am I testing?" conditionals inside non-test files. Everything test related (mocks for example) should be assembled/injected from test suites, as long as the design is flexible enough to allow it.